internal/agentdeployer/_static/docker-agent-base.yml.tmpl

```diff
@@ -1,35 +1,43 @@
+{{- $user := fact "user" -}}
+{{- $capabilities:= fact "capabilities" -}}
+{{- $pid_mode := fact "pid_mode" -}}
+{{- $ports := fact "ports" -}}
+{{- $custom_script:= fact "custom_script" -}}
 services:
   elastic-agent:
     hostname: ${AGENT_HOSTNAME}
+    {{ if ne $custom_script "" }}
+    image: "${ELASTIC_AGENT_IMAGE_REF}-custom"
```

Comment:
Added as a suffix the MD5 sum of the Dockerfile created.
image: "${ELASTIC_AGENT_IMAGE_REF}-ep-{{ $dockerfile_hash }}"

The result is images like this:
```
docker.elastic.co/elastic-agent/elastic-agent-complete   8.13.2-ep-0b62e54b357d3c48c702361c4dbb8925   f9873a3ecbda   3 minutes ago       3.95GB
docker.elastic.co/elastic-agent/elastic-agent-complete   8.13.2-ep-6f2ddf69e9db6e8f543524e9c91f483b   7ce72492afd1   56 minutes ago      3.61GB
```

Maybe it could be added also `-ep`